metadata-ingestion/src/datahub/sql_parsing/sqlglot_lineage.py

```diff
@@ -1258,6 +1275,20 @@ def infer_output_schema(result: SqlParsingResult) -> Optional[List[SchemaFieldCl
     return output_schema
 
 
+def infer_upstream_columns(
```

Comment:
this method name is pretty confusing - maybe match_columns_to_schema?

also `expected_columns` is also a confusing way to name this

Reply:
done